dotenv-linter/src/checks/schema_validator.rs

```diff
@@ -0,0 +1,117 @@
+use std::collections::HashSet;
+
+use super::Check;
+use crate::{
+    common::{LintKind, Warning},
+    schema::SchemaValueType,
+};
+use dotenv_lookup::LineEntry;
+use email_address::EmailAddress;
+use url::Url;
+pub(crate) struct SchemaChecker<'a> {
+    schema: Option<&'a crate::schema::DotEnvSchema>,
+    seen_keys: HashSet<String>,
+    last_line_number: usize,
+}
+
+impl<'a> SchemaChecker<'a> {
+    pub fn new(opts: &'a crate::cli::options::CliOptions) -> Self {
+        Self {
+            schema: opts.schema.as_ref(),
+            seen_keys: HashSet::new(),
+            last_line_number: 0,
+        }
+    }
+}
+
+impl Check for SchemaChecker<'_> {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        let schema = self.schema?;
+        self.last_line_number = line.number;
+        let key = line.get_key()?;
+        self.seen_keys.insert(key.to_string());
+
+        if let Some(entry) = schema.entries.get(key) {
+            if let Some(value) = line.get_value() {
+                match entry.value_type {
+                    SchemaValueType::String => {
+                        if let Some(regex) = &entry.regex {
+                            if !regex::Regex::new(regex).unwrap().is_match(value) {
+                                return Some(Warning::new(
+                                    line.number,
+                                    self.name(),
+                                    format!("The {} key does not match the regex", key),
+                                ));
+                            }
+                        }
+                    }
+                    SchemaValueType::Boolean => {
+                        if value == "true" || value == "false" {
+                            return None;
+                        } else {
+                            return Some(Warning::new(
+                                line.number,
+                                self.name(),
+                                format!("The {} key is not a valid boolean", key),
+                            ));
+                        }
+                    }
```

Comment:
I guess for boolean we also would need to think about possible validations.

Because even `clap` provide couple of parsers: 
- [BoolishValueParser](https://docs.rs/clap/latest/clap/builder/struct.BoolishValueParser.html)
- [FalseyValueParser](https://docs.rs/clap/latest/clap/builder/struct.FalseyValueParser.html)

And what to consider `true` or `false` is responsibility of application which works with that. For now we will have hard-check 🤔 

So probably we should consider passing lists of supported values.

For first version I think it's ok to stick to `true` & `false`.  I'd say lower & upper case.

Reply:
booleans can be done via regex too

Reply:
I have updated the boolean test